components/prediction-page/prediction-page.tsx

```diff
@@ -1,91 +1,321 @@
 import {
+  _,
   abbreviateNumber,
+  Computed,
   formatNumber,
-  gql,
   Icon,
   ImageByAspectRatio,
+  Memo,
+  Observable,
   React,
-  usePollingQuery,
+  useEffect,
+  useMutation,
+  useObserve,
+  useSelector,
+  useSignal,
   useStyleSheet,
+  useUrqlQuerySignal,
 } from "../../deps.ts";
-import { CRYSTAL_BALL_ICON, CRYSTAL_BALL_ICON_VIEWBOX } from "../../shared/mod.ts";
+import {
+  CRYSTAL_BALL_ICON,
+  CRYSTAL_BALL_ICON_VIEWBOX,
+  DELETE_POLL_MUTATION,
+  hasPermission,
+  useOrgUserWithRoles$,
+} from "../../shared/mod.ts";
+import { ChannelPoints, Poll } from "../../types/mod.ts";
+import Prediction from "../prediction/prediction.tsx";
+import { ACTIVE_PREDICTION_QUERY, CHANNEL_POINTS_QUERY, POLL_QUERY } from "../prediction/gql.ts";
 import { Page, usePageStack } from "../page-stack/mod.ts";
-
-import ActivePrediction from "../active-prediction/active-prediction.tsx";
+import Button from "../base/button/button.tsx";
 import styleSheet from "./prediction-page.scss.js";
 
-const CHANNEL_POINTS_QUERY = gql`
-  query ChannelPointsQuery {
-    channelPoints: orgUserCounterType(input: { slug: "channel-points" }) {
-      orgUserCounter {
-        count
-      }
+const CHANNEL_POINTS_SRC =
+  "https://cdn.bio/assets/images/features/browser_extension/channel-points-default.svg";
+
+function usePollingChannelPoints$(
+  { interval = 1000 }: { interval?: number },
+) {
+  const channelPoints$ = useSignal<ChannelPoints>(undefined!);
+  const { signal$: channelPointsData$, reexecuteQuery: reexecuteChannelPointsQuery } =
+    useUrqlQuerySignal(
+      CHANNEL_POINTS_QUERY,
+    );
+
+  useEffect(() => {
+    const id = setInterval(() => {
+      reexecuteChannelPointsQuery({ requestPolicy: "network-only" });
+    }, interval);
+
+    return () => clearInterval(id);
+  }, []);
+
+  // we use this to cut down rerenders in the prediction component by only updating the observable
+  // if the response has changed vs. on every interval
+  useObserve(() => {
+    const channelPoints = channelPointsData$.data?.get()?.channelPoints.orgUserCounter;
+    const existingChannelPoints = channelPoints$.get();
+    if (channelPoints && !_.isEqual(channelPoints, existingChannelPoints)) {
+      channelPoints$.set(channelPoints);
+    }
+  });
+
+  return { channelPoints$, reexecuteChannelPointsQuery };
+}
+
+function usePollingPrediction$(
+  { interval = 2000, pollId }: { interval?: number; pollId: string },
+) {
+  const prediction$ = useSignal<Poll>(undefined!);
+
+  const { signal$: predictionResponse$, reexecuteQuery: reexecutePredictionQuery } =
+    useUrqlQuerySignal(
+      POLL_QUERY,
+      {
+        id: pollId,
+      },
+    );
+
+  useEffect(() => {
+    const id = setInterval(() => {
+      reexecutePredictionQuery({ requestPolicy: "network-only" });
```

Comment:
looks like all our polling queries have a lot of overlap in code, can we pull that out?

Reply:
https://github.com/trufflehq/truffle-packages/pull/42